crates/runtime/src/flight/do_put.rs

```diff
@@ -69,16 +73,27 @@ pub(crate) async fn handle(
         }
     }
 
-    let mut streaming_flight = request.into_inner();
+    let mut streaming_flight = request.into_inner().peekable();
 
-    let Ok(Some(message)) = streaming_flight.message().await else {
+    let Some(Ok(first_message)) = streaming_flight.peek().await else {
         let _start = metrics::track_flight_request("do_put", None);
         return Err(Status::invalid_argument("No flight data provided"));
     };
-    let Some(fd) = &message.flight_descriptor else {
+    let Some(fd) = &first_message.flight_descriptor else {
         let _start = metrics::track_flight_request("do_put", None);
         return Err(Status::invalid_argument("No flight descriptor provided"));
     };
+
+    if let Ok(message) = Any::decode(&*fd.cmd) {
+        match Command::try_from(message).map_err(|e| Status::internal(format!("{e:?}")))? {
+            Command::CommandPreparedStatementQuery(query) => {
+                return prepared_statement_query::do_put_query(flight_svc, query, streaming_flight)
+                    .await;
+            }
+            c => return Err(Status::unimplemented(&format!("{c:?}"))),
```

Comment:
Please add a note why we always return error here and not executing `do_put` further